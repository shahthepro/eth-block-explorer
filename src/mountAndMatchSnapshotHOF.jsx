export default function (it, ReactDOM, mount) {
  return function (elementName, element) {
    it(`renders <${elementName}> without crashing`, () => {
      const div = document.createElement('div');
      ReactDOM.render(element, div);
      ReactDOM.unmountComponentAtNode(div);
    });
    
    it(`matches <${elementName}> snapshot`, () => {
      const mountedElement = mount(element);
      expect(mountedElement).toMatchSnapshot();
    });
  }
}