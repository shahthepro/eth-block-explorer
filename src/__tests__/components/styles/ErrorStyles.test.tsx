import * as React from 'react';
import * as ReactDOM from 'react-dom';
import {
  ErrorMessage,
  InfoMessage
} from './../../../components/styles/ErrorStyles';
import * as Enzyme from 'enzyme';
import * as Adapter from 'enzyme-adapter-react-16';
import mountAndMatchSnapshotHOF from '../../../../mountAndMatchSnapshotHOF';

Enzyme.configure({ adapter: new Adapter() })
const { mount } = Enzyme;

const mountAndMatchSnapshot = mountAndMatchSnapshotHOF(it, ReactDOM, mount);

mountAndMatchSnapshot('ErrorMessage', <ErrorMessage />);
mountAndMatchSnapshot('InfoMessage', <InfoMessage />);