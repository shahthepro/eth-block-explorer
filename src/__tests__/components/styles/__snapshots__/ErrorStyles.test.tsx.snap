// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches <ErrorMessage> snapshot 1`] = `
<Styled(styled.div)>
  <StyledComponent
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-htpNat",
          "isStatic": true,
          "lastClassName": "jwBQwt",
          "rules": Array [
            "
        padding: 1rem;
        background: white;
        margin: 1rem 0;
        border: 1px solid rgba(0, 0, 0, 0.05);
        border-left: 5px solid red;
        p {
          margin: 0;
          font-weight: 100;
        }
        strong {
          margin-right: 1rem;
        }
      ",
            "
        border-left: 5px solid red;
      ",
          ],
        },
        "displayName": "Styled(styled.div)",
        "foldedComponentIds": Array [
          "sc-bdVaJa",
        ],
        "render": [Function],
        "styledComponentId": "sc-htpNat",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
  >
    <div
      className="sc-bdVaJa sc-htpNat jwBQwt"
    />
  </StyledComponent>
</Styled(styled.div)>
`;

exports[`matches <InfoMessage> snapshot 1`] = `
<Styled(styled.div)>
  <StyledComponent
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bwzfXH",
          "isStatic": true,
          "lastClassName": "kdOuOG",
          "rules": Array [
            "
        padding: 1rem;
        background: white;
        margin: 1rem 0;
        border: 1px solid rgba(0, 0, 0, 0.05);
        border-left: 5px solid red;
        p {
          margin: 0;
          font-weight: 100;
        }
        strong {
          margin-right: 1rem;
        }
      ",
            "
        border-left: 5px solid rgb(0, 122, 255);
      ",
          ],
        },
        "displayName": "Styled(styled.div)",
        "foldedComponentIds": Array [
          "sc-bdVaJa",
        ],
        "render": [Function],
        "styledComponentId": "sc-bwzfXH",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
  >
    <div
      className="sc-bdVaJa sc-bwzfXH kdOuOG"
    />
  </StyledComponent>
</Styled(styled.div)>
`;
